packages/mobile/src/utils/time.ts

```diff
@@ -318,3 +320,29 @@ function quickFormat(timestamp: number, i18next: i18nType, formatRule: string) {
     locale: i18next.language.includes('es') ? es : enUS,
   })
 }
+
+// update moving average for time ellapsed
+export function updateMeanMillisecs(meanMillisecs: number, millisecs: number, onceEvery: number) {
+  const multiplier = meanMillisecs > 0.0 ? meanMillisecs / onceEvery : 1.0
+  meanMillisecs = meanMillisecs * (1.0 - multiplier) + multiplier * millisecs
+  return meanMillisecs
+}
+
+// warning: this function has side-effects on module
```

Comment:
I think this abstraction could be improved so that you don't mutate external vars in the function. If you create an object in this file with keys that are the event name and values that store data, you could track meanMilliseconds and trackCount there

Reply:
Okey, got it, thanks.